DigitalLearningSolutions.Data/Helpers/DateHelper.cs

```diff
@@ -1,12 +1,68 @@
-﻿namespace DigitalLearningSolutions.Web.Helpers
+﻿namespace DigitalLearningSolutions.Data.Helpers
 {
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using DigitalLearningSolutions.Data.Enums;
 
     public static class DateHelper
     {
-        public static IEnumerable<(int Month, int Year)> GetMonthsAndYearsBetweenDates(DateTime startDate, DateTime endDate)
+        public static DateTime ReferenceDate => new DateTime(1905, 1, 1);
+
+        public static IEnumerable<DateTime> GetPeriodsBetweenDates(
+            DateTime startDate,
+            DateTime endDate,
+            ReportInterval interval
+        )
+        {
+            return interval switch
+            {
+                ReportInterval.Days => GetDaysBetweenDates(startDate, endDate),
+                ReportInterval.Weeks => GetWeeksBetweenDates(startDate, endDate),
+                ReportInterval.Months => GetMonthsBetweenDates(startDate, endDate),
+                ReportInterval.Quarters => GetQuartersBetweenDates(startDate, endDate),
+                _ => GetYearsBetweenDates(startDate, endDate)
+            };
+        }
+
+        private static IEnumerable<DateTime> GetDaysBetweenDates(
+            DateTime startDate,
+            DateTime endDate
+        )
+        {
+            var diffInDays = (endDate - startDate).Days;
+            var daysDifferenceEnumerable = Enumerable.Range(0, diffInDays + 1);
+
+            return daysDifferenceEnumerable.Select(
+                d =>
+                {
+                    var day = startDate.AddDays(d);
+                    return new DateTime(day.Year, day.Month, day.Day);
+                }
+            );
+        }
+
+        private static IEnumerable<DateTime> GetWeeksBetweenDates(
+            DateTime startDate,
+            DateTime endDate
+        )
+        {
+            var referenceDate = ReferenceDate;

```

Comment:
This is a minor optimisation. ReferenceDate, to be static, has to be a call to get a new DateTime rather than just a DateTime (as I understand it). This means the same datetime doesn't have to be recreated over and over again inside the Select loops.